wandb/sdk/artifacts/artifacts_cache.py

```diff
@@ -181,6 +181,13 @@ def helper(mode: str = "w") -> Generator[IO, None, None]:
             try:
                 yield temp_file
                 temp_file.close()
+                # The NamedTemporaryFile is always created with mode 0600, hardcoded at
+                # tempfile.py, line 235, because it is private to your process until
+                # you open it up with chmod. Here we change the permissions back to
+                # the system defaults, with a temporary umask assignment to 0o022
+                umask = os.umask(0o22)
+                os.umask(umask)
```

Comment:
Nit:
- let's just point to the StackOverflow discussion
- it wasn't clear from this comment why you set the umask twice
```suggestion
                # NamedTemporaryFile sets the file mode to 600 [1], we reset to the default.
                # [1] https://stackoverflow.com/questions/10541760/can-i-set-the-umask-for-tempfile-namedtemporaryfile-in-python
                umask = os.umask(0o22)  # we have to set umask to get the current value
                os.umask(umask)  # reset
```